NPE in ElasticsearchExceptionTranslator

Hi,

I'm migrating from Spring Boot 2.7 to Spring Boot 3 and switching to new Spring Data Elasticsearch.

I have many problems with it (most of them with rewriting queries and aggregations from old to new Spring Data - since `QueryBuilders` class from here is missing queries, I must mix it with `QueryBuilders` from ES client and direct builders for query/aggregations from ES client -  it is not the happiest solution that the  `QueryBuilders` class in Spring Data Elasticsearch is named the same as in the ES client, and at many places I need to use `._toQuery()` from `QueryVariant`)

Back to the topic, `ElasticsearchExceptionTranslator` is generating `NullPointerException` because `ErrorResponse.error().type() is null`:
![image](https://user-images.githubusercontent.com/1428972/205597423-2c0dd58c-3070-4dc5-b1b0-9cba898a19b4.png)


![image](https://user-images.githubusercontent.com/1428972/205595564-bc5bb1b3-984a-45ff-9bbd-ea9fce06fc98.png)


```log
Caused by: java.lang.NullPointerException: Cannot invoke "String.contains(java.lang.CharSequence)" because the return value of "co.elastic.clients.elasticsearch._types.ErrorCause.type()" is null
	at org.springframework.data.elasticsearch.client.elc.ElasticsearchExceptionTranslator.translateExceptionIfPossible(ElasticsearchExceptionTranslator.java:91)
	at org.springframework.data.elasticsearch.client.elc.ElasticsearchExceptionTranslator.translateException(ElasticsearchExceptionTranslator.java:62)
	at org.springframework.data.elasticsearch.client.elc.ChildTemplate.execute(ChildTemplate.java:73)
```

--- a/src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchExceptionTranslator.java
+++ b/src/main/java/org/springframework/data/elasticsearch/client/elc/ElasticsearchExceptionTranslator.java
@@ -74,13 +74,15 @@ public class ElasticsearchExceptionTranslator implements PersistenceExceptionTra
 		if (ex instanceof ElasticsearchException elasticsearchException) {
 
 			ErrorResponse response = elasticsearchException.response();
+			var errorType = response.error().type();
+			var errorReason = response.error().reason() != null ? response.error().reason() : "undefined reason";
 
-			if (response.status() == 404 && "index_not_found_exception".equals(response.error().type())) {
+			if (response.status() == 404 && "index_not_found_exception".equals(errorType)) {
 
 				// noinspection RegExpRedundantEscape
 				Pattern pattern = Pattern.compile(".*no such index \\[(.*)\\]");
 				String index = "";
-				Matcher matcher = pattern.matcher(response.error().reason());
+				Matcher matcher = pattern.matcher(errorReason);
 				if (matcher.matches()) {
 					index = matcher.group(1);
 				}
@@ -88,8 +90,8 @@ public class ElasticsearchExceptionTranslator implements PersistenceExceptionTra
 			}
 			String body = JsonUtils.toJson(response, jsonpMapper);
 
-			if (response.error().type().contains("validation_exception")) {
-				return new DataIntegrityViolationException(response.error().reason());
+			if (errorType != null && errorType.contains("validation_exception")) {
+				return new DataIntegrityViolationException(errorReason);
 			}
 
 			return new UncategorizedElasticsearchException(ex.getMessage(), response.status(), body, ex);

